### PR TITLE
UI Nitpicks

### PR DESCRIPTION
**Pull request by wo0dyn**
## TL;DR

**Welcome emojis!** 🎉  
I kept emoticons on the API view, even if emojis are valid in a JSON response.

## Screenshots

Screen Shot 2020-12-04 at 11 27 32 AM
Screen Shot 2020-12-04 at 11 27 57 AM

## ChangeLog

* Switch templates to 2 spaces indentation
* Remove unrelated class to <html> element
* HTML5-ize template markup
* Use url() function from Symfony
* Remove `Action` suffix on controller's methods
* Return JSON content for console-based browser
* Fix typo in comment
* 2020-ize the main and subtext messages with emojis
* Improve readibility of the code using date() options
* Switch simple → double-quotes

## Review

To be reviewed commit by commit. Feel free to tell me if some of them might be dropped.

## Note 

The last version of symfony I used was the 1.4 — Hence the lower `s`. So be gentle with me if I didn't follow the latest standard coding style from the PHP community. 🙏  

Thanks to @fouick for the wording.

### CODE REVIEW

**Review by damienalexandre — APPROVED** [review]
Damn nice :+1:

File: application/src/Weekend.php [hunk 1]
```diff
@@ -4,49 +4,81 @@
...
 {
+    protected function getMapping()
+    {
+        $mapping = [
+            '\o/'  => '🎉',
+            '◄:•D' => '🥳',
+            'B-)'  => '😎',
+            '¬‿¬'  => '😏',
+            ':('   => '😟',
+            'X-('  => '😠',
```
Comment (pyrech) [inline application/src/Weekend.php:15]: You could also add a mapping for `¯\_(ツ)_/¯` => `🤷`

File: application/src/Weekend.php [hunk 2]
```diff
@@ -4,49 +4,81 @@
 
 class Weekend
 {
+    protected function getMapping()
+    {
+        $mapping = [
```
Comment (pyrech) [inline application/src/Weekend.php:9]: this map could be a const instead of this method

const MAPPING => [...];

It would then be used with `self::MAPPING` instead of `$this->getMapping()`

**Review by pyrech — APPROVED** [review]

**Comment by fouick** [comment]
"¯\_(ツ)_/¯ => 🤷"

La conversion "bonhomme qui hausse les épaules" ne rend pas hommage au shrug officiel sur-utilisé par Ternel 🙂
On a essayé de respecter la légende!

</avis perso>



Elise Clay | Director Weekend

________________________________
De : Loïck Piera <notifications@github.com>
Envoyé : vendredi 4 décembre 2020 15:31
À : ternel/weekend-website <weekend-website@noreply.github.com>
Cc : Elise Clay <elise.clay@ukg.com>; Mention <mention@noreply.github.com>
Objet : [External Origin] Re: [ternel/weekend-website] UI Nitpicks (#19)


@pyrech approved this pull request.

________________________________

In application/src/Weekend.php<https://github.com/ternel/weekend-website/pull/19#discussion_r536138130>:

> @@ -4,49 +4,81 @@



 class Weekend

 {

+    protected function getMapping()

+    {

+        $mapping = [

+            '\o/'  => '🎉',

+            '◄:•D' => '🥳',

+            'B-)'  => '😎',

+            '¬‿¬'  => '😏',

+            ':('   => '😟',

+            'X-('  => '😠',



You could also add a mapping for ¯\_(ツ)_/¯ => 🤷

________________________________

In application/src/Weekend.php<https://github.com/ternel/weekend-website/pull/19#discussion_r536139106>:

> @@ -4,49 +4,81 @@



 class Weekend

 {

+    protected function getMapping()

+    {

+        $mapping = [



this map could be a const instead of this method

const MAPPING => [...];

It would then be used with self::MAPPING instead of $this->getMapping()

**Comment by pyrech** [comment]
C'est un argument parfaitement recevable :+1:

**Comment by ternel** [comment]
¯\_(ツ)_/¯

**Comment by ternel** [comment]
I would like to welcome @wo0dyn, @lyrixx and @pyrech as core contributor. (Some because they contribute... some because there are jealous.)


Soon, the blogpost.
Screen Capture_select-area_20201204155924

**Comment by wo0dyn** [comment]
Prior to contributing, I didn't know the project was _**that**_ famous. 😳  

image

File: application/src/Weekend.php (hunk 2 above)
Comment (wo0dyn) [reply application/src/Weekend.php]: Done in c85d9b77c7293c186477cb39d7d367c4ffeff31b.
Does that look good to you?

File: application/src/Weekend.php (hunk 1 above)
Comment (wo0dyn) [reply application/src/Weekend.php]: See https://github.com/ternel/weekend-website/pull/19#issuecomment-738824364 😉

File: application/src/Weekend.php (hunk 2 above)
Comment (pyrech) [reply application/src/Weekend.php]: Absolutely perfect! Well done :ok_hand:

**Review by pyrech — APPROVED** [review]
Perfect! Let's ship it! :tada:

File: application/src/Controller/DefaultController.php [hunk 3]
```diff
@@ -8,11 +8,17 @@
 
 class DefaultController extends Controller
 {
-    public function indexAction(Weekend $weekend)
+    public function index(Weekend $weekend)
     {
+        if(!preg_match('/AppleWebKit|Chrome|Edge|Gecko|Opera|Trident/i', $_SERVER['HTTP_USER_AGENT'])) {
```
Comment (lyrixx) [inline application/src/Controller/DefaultController.php:13]:
```suggestion
    public function index(Weekend $weekend, Request $request)
    {
        if(!preg_match('/AppleWebKit|Chrome|Edge|Gecko|Opera|Trident/i', $request->headers->get('User-Agent'))) {
```
And you'll have to add the `use ...\Request`

I did not try the code, but it should be something similar

File: application/src/Weekend.php [hunk 4]
```diff
@@ -4,49 +4,77 @@
 
 class Weekend
 {
+    const MAPPING = [
```
Comment (lyrixx) [inline application/src/Weekend.php:7]:
```suggestion
    private const MAPPING = [
```

**Comment by bastnic** [comment]
Congrats @wo0dyn, @lyrixx and @pyrech. Hi @fouick!

File: application/src/Weekend.php [hunk 5]
```diff
@@ -4,49 +4,77 @@
...
+        ':('   => '😟',
+        'X-('  => '😠',
+    ];
+
+    /**
+     * Compute the main text with emojis
+     *
+     * @return string
+     */
+    public function getRichText()
```
Comment (lyrixx) [inline application/src/Weekend.php:21]:
```suggestion
    /**
     * Compute the main text with emojis
     */
    public function getRichText(): string
```

File: application/src/Weekend.php [hunk 6]
```diff
@@ -4,49 +4,77 @@
...
+    {
+        $emoticons = array_keys(self::MAPPING);
+        $emojis = array_values(self::MAPPING);
+
+        return str_replace($emoticons, $emojis, $this->getText());
```
Comment (lyrixx) [inline application/src/Weekend.php:26]:
```suggestion
        return strtr($emojis, $this->getText(), self::MAPPING);
```

File: application/src/Weekend.php [hunk 7]
```diff
@@ -4,49 +4,77 @@
...
+    }
+
+    /**
+     * Compute the main subtext with emojis
+     *
+     * @return string
+     */
+    public function getRichSubText()
```
Comment (lyrixx) [inline application/src/Weekend.php:34]:
```suggestion
     * Compute the main subtext with emojis
     */
    public function getRichSubText(): string
```

File: application/src/Weekend.php [hunk 8]
```diff
@@ -4,49 +4,77 @@
...
+    {
+        $emoticons = array_keys(self::MAPPING);
+        $emojis = array_values(self::MAPPING);
+
+        return str_replace($emoticons, $emojis, $this->getSubText());
```
Comment (lyrixx) [inline application/src/Weekend.php:39]:
```suggestion
        return strtr($this->getSubText(), self::MAPPING);
```

File: application/src/Weekend.php [hunk 9]
```diff
@@ -4,49 +4,77 @@
...
+    }
+
     /**
      * Compute the main text
      *
      * @return string
      */
     public function getText()
```
Comment (lyrixx) [inline application/src/Weekend.php:47]:
```suggestion
     */
    public function getText(): string
```

**Review by lyrixx — COMMENTED** [review]
That's a very good PR. Thanks a lot.
I left some minor comments :) Feel free to accept them (or not :) )

File: application/src/Controller/DefaultController.php (hunk 3 above)
Comment (wo0dyn) [reply application/src/Controller/DefaultController.php]: Done, thanks! 👌

File: application/src/Weekend.php (hunk 4 above)
Comment (wo0dyn) [reply application/src/Weekend.php]: 👌

File: application/src/Weekend.php (hunk 5 above)
Comment (wo0dyn) [reply application/src/Weekend.php]: Thanks, I've done this on other methods: `getText()` and `getSubText()`.

File: application/src/Weekend.php (hunk 6 above)
Comment (wo0dyn) [reply application/src/Weekend.php]: Nice! I didn't recall this function, but that's what I was looking for indeed. 👍

**Comment by wo0dyn** [comment]
> I left some minor comments :) Feel free to accept them (or not :) )

Thanks for the review. Everything has been addressed in d77bcdf2edb990494ed76a0239b8a4a8a823b34e. You've been given credits, as well @pyrech (I forgot on the previous fixup). 👍

**Review by ternel — APPROVED** [review]

**Comment by ternel** [comment]
After a downtime of ~5min... https://estcequecestbientotleweekend.fr/
